File: 3-longest-substring-without-repeating-characters/brute-force/brute-force.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        maxLen = 0
        i = 0

        while (i < len(s)):
            count = [0] * 128
            j = i
            while (j < len(s)):
                # Get ASCII value of s[j]
                code = ord(s[j])
                if (count[code] != 0):
                    break
                else:
                    maxLen = max(maxLen, (j - i + 1))
                    count[code] = 1
                j += 1
            i += 1

        return maxLen

File: 3-longest-substring-without-repeating-characters/brute-force/brute-force.cpp
class Solution
{
    public:

        int lengthOfLongestSubstring(string s)
        {
            int len = s.length();
            int max = 0;

            for (int i = 0; i < len; i++)
            {
                int count[128] = {0};

                for (int j = i; j < len; j++)
                {
                    if (count[s[j]])
                    {
                        break;
                    }
                    else
                    {
                        max = std::max(max, (j - i + 1));
                        count[s[j]] = 1;
                    }
                }
            }

            return max;
        }
};